### ISSUE TITLE
Function return struct

### ISSUE THREAD

**Issue by BenShen98** [LangProc/langproc-2018-cw#26]
In the test case, are we required to return a `struct`? And do we have to follow the ABI conversion to use the first arguement register to provide the reserved space for such returned struct?

If we need to return the `struct`, would such test case also contain `typedef`?

**Comment by ymherklotz** [comment]
I don't believe that there is a required test case for `struct`, but if you do want your compiler to support it, you will have to follow the ABI conventions so that it can be called and retrieved by GCC.

`struct` also does not need `typedef`, it often just makes it easier to read.
